ios/RNDateTimePickerShadowView.m

```diff
@@ -0,0 +1,52 @@
+#import "RNDateTimePickerShadowView.h"
+
+@implementation RNDateTimePickerShadowView
+
+- (instancetype)init
+{
+  if (self = [super init]) {
+    YGNodeSetMeasureFunc(self.yogaNode, RNDateTimePickerShadowViewMeasure);
+  }
+  return self;
+}
+
+- (void)setDate:(NSDate *)date {
+  _date = date;
+  YGNodeMarkDirty(self.yogaNode);
+}
+
+- (void)setLocale:(NSLocale *)locale {
+  _locale = locale;
+  YGNodeMarkDirty(self.yogaNode);
+}
+
+- (void)setMode:(UIDatePickerMode)mode {
+  _mode = mode;
+  YGNodeMarkDirty(self.yogaNode);
+}
+
+static YGSize RNDateTimePickerShadowViewMeasure(YGNodeRef node, float width, YGMeasureMode widthMode, float height, YGMeasureMode heightMode)
+{
+  RNDateTimePickerShadowView *shadowPickerView = (__bridge RNDateTimePickerShadowView *)YGNodeGetContext(node);
+  
+  dispatch_group_t group = dispatch_group_create();
+  __block CGSize size;
+  dispatch_group_async(group, dispatch_get_main_queue(), ^{
```

Comment:
This can have negative performance implications since we execute an asynchronous call synchronously. This is because we need to call `setPreferredDatePickerStyle:shadowPickerView` and `sizeThatFits`, both of which need to be executed on the main thread – whereas the measure function is called from a non-main thread. Personally, I think it's better to have some performance impact rather than a wrong layout but if anybody has a better idea how to get around this limitation, let me know.